api/src/main/java/org/datacleaner/metadata/DefaultColumnMeaningCollection.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * DataCleaner (community edition)
+ * Copyright (C) 2014 Neopost - Customer Information Management
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.metadata;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class DefaultColumnMeaningCollection implements ColumnMeaningCollection {
+    private static Map<String, HasColumnMeaning> _matchingMap;
+
+    static {
+        _matchingMap = new HashMap<>();
+        final HasColumnMeaning[] values = ColumnMeaning.class.getEnumConstants();
+
+        for (final HasColumnMeaning columnMeaning : values) {
+            populateMatchMap(columnMeaning.getName(), columnMeaning);
+
+            for (final String alias : columnMeaning.getAliases()) {
+                populateMatchMap(alias, columnMeaning);
+            }
+        }
+    }
```

Comment:
Such a big static initialiser is a little bit nasty. Can we do this lazily instead?